### ISSUE TITLE
Ubuntu 18.04 install instruction on README clash with amdgpu driver

### ISSUE THREAD

**Issue by smoe** [RadeonOpenCompute/ROCm#1044]
Hello, following the instructions of this repository's README I ran into
```
Unpacking rock-dkms (3.1-44) ...
dpkg: error processing archive /var/cache/apt/archives/rock-dkms_3.1-44_all.deb (--unpack):
 trying to overwrite '/usr/share/dkms/modules_to_force_install/amdgpu', which is also in package amdgpu-dkms 19.50-967956
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 /var/cache/apt/archives/rock-dkms_3.1-44_all.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```
Since the amdgpu-dkms is the way us users are guided to install starting out from the amd.com home page, I presume that this error message will be experienced by many others, too. Please kindly prepare your instructions for that - or (preferably) allow the two packagse to be coinstalled.

Otherwise, that far, the installation went very smoothly.

**Comment by AndrewCarterUK** [comment]
I think if you install the ROCm drivers you don't need the amd-gpu driver (as it is based off them).

Not 100% sure but I encountered something similar and now I just have the ROCm drivers installed.

**Comment by smoe** [comment]
Thank you, @AndrewCarterUK. I just tried it. It seems like the rock-dkms is upward-compatible with amdgpu-dkms.
sudo dpkg --purge amdgpu-dkms
followed by
sudo apt-get install rock-dkms
fixed this. I was running BOINC jobs of Einstein@Home who had failed when I remove the kernel module. With a reboot after the installation of rock-dkms they just ran again.

The Debian/Ubuntu package should possibly state 
```
Provides: amdgpu-dkms (so nothing needs to get uninstalled)
Conflicts: amdgpu-dkms (so the existing module goes away when the rock-dkms gets installed)
```
in its debian/control file. And, well, I just hope to help getting the installl instructions even more foolproof.

**Comment by smoe** [comment]
The patch described on
https://github.com/RadeonOpenCompute/ROCm/pull/1048
I expect to help others who are up for the transition.

**Comment by Wedge009** [comment]
Hello @smoe, were you able to get ROCm running with BOINC? I'm a fellow E@H contributor and since amdgpu-pro 20.45 onwards has switched from PAL-based OpenCL to ROCr, _all_ BOINC GPU processing now fails (looks like cannot compile any OpenCL kernels despite OpenCL being detected.

I did try ROCm v3.x some time ago but never managed to get BOINC to recognise OpenCL from it, so I gave up and reverted to amdgpu-pro.

**Comment by smoe** [comment]
Hello @Wedge009, yes, https://einsteinathome.org/de/host/12784717 crunches happily with ROCm on RX580s. At least so I thought it was. This is what I have installed:

```
moeller@rig1c:~$ dpkg -l "*rocm*"
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                  Version                             Architecture Description
+++-=====================-===================================-============-====================================================================================================================>
un  procmail              <none>                              <none>       (no description available)
ii  rocm-clang-ocl        0.5.0.64-rocm-rel-4.0-23-50fb51a    amd64        OpenCL compilation with clang compiler.
ii  rocm-cmake            0.3.0.153-rocm-rel-4.0-23-1d1caa5   amd64        rocm-cmake built using CMake
ii  rocm-dbgapi           0.42.0.40000-23                     amd64        Library to provide AMD GPU debugger API
ii  rocm-debug-agent      2.0.1.40000-23                      amd64        Radeon Open Compute Debug Agent (ROCdebug-agent)
ii  rocm-dev              4.0.0.40000-23                      amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-device-libs      1.0.0.637-rocm-rel-4.0-23-db8c0c3   amd64        Radeon Open Compute - device libraries
ii  rocm-dkms             4.0.0.40000-23                      amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-gdb              10.1-rocm-rel-4.0-23                amd64        ROCgdb
ii  rocm-libs             4.0.0.40000-23                      amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-opencl           3.6Beta-17-g875c1f8-rocm-rel-4.0-23 amd64        OpenCL: Open Computing Language on ROCclr
ii  rocm-opencl-dev       3.6Beta-17-g875c1f8-rocm-rel-4.0-23 amd64        OpenCL: Open Computing Language on ROCclr
ii  rocm-smi              1.0.0-206-rocm-rel-4.0-23-ge39c0e2  amd64        System Management Interface for ROCm
ii  rocm-smi-lib64        2.9.0.9-rocm-rel-4.0-23-4b49d2d     amd64        AMD System Management libraries
ii  rocm-utils            4.0.0.40000-23                      amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-validation-suite 3.10.40000                          amd64        The ROCm Validation Suite – The ROCm Validation Suite is a system administrator and cluster manager's tool for detec>
ii  rocminfo              1.40000.0                           amd64        Radeon Open Compute (ROCm) Runtime rocminfo tool
lines 1-22/22 (END)
dpkg -l *amdgpu*
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                Version         Architecture Description
+++-===================================-===============-============-=====================================================================
ii  amdgpu-core                         19.50-967956    all          Core meta package for unified amdgpu driver.
ii  amdgpu-pro-core                     19.50-967956    all          Core meta package for Pro components of the unified amdgpu driver.
ii  amdgpu-pro-pin                      19.50-967956    all          Meta package to pin a specific amdgpu driver version.
ii  clinfo-amdgpu-pro                   19.50-967956    amd64        AMD OpenCL info utility
ii  libdrm-amdgpu-amdgpu1:amd64         1:2.4.99-967956 amd64        Userspace interface to amdgpu-specific kernel DRM services -- runtime
ii  libdrm-amdgpu-common                1.0.0-967956    all          List of AMD/ATI cards' device IDs, revision IDs and marketing names
ii  libdrm-amdgpu1:amd64                2.4.101-2       amd64        Userspace interface to amdgpu-specific kernel DRM services -- runtime
ii  libdrm2-amdgpu:amd64                1:2.4.99-967956 amd64        Userspace interface to kernel DRM services -- runtime
ii  libllvm9.0-amdgpu:amd64             1:9.0-967956    amd64        Modular compiler and toolchain technologies, runtime library
ii  libopencl1-amdgpu-pro:amd64         19.50-967956    amd64        AMD OpenCL ICD Loader library
ii  llvm-amdgpu                         1:9.0-967956    amd64        Low-Level Virtual Machine (LLVM)
ii  llvm-amdgpu-9.0                     1:9.0-967956    amd64        Modular compiler and toolchain technologies
ii  llvm-amdgpu-9.0-dev                 1:9.0-967956    amd64        Modular compiler and toolchain technologies, libraries and headers
ii  llvm-amdgpu-9.0-runtime             1:9.0-967956    amd64        Modular compiler and toolchain technologies, IR interpreter
ii  llvm-amdgpu-runtime                 1:9.0-967956    amd64        Low-Level Virtual Machine (LLVM), bytecode interpreter
ii  opencl-amdgpu-pro-comgr             19.50-967956    amd64        non-free AMD OpenCL ICD Loaders
ii  opencl-amdgpu-pro-icd               19.50-967956    amd64        non-free AMD OpenCL ICD Loaders
ii  opencl-orca-amdgpu-pro-icd:amd64    19.50-967956    amd64        non-free AMD OpenCL ICD Loaders
ii  ricks-amdgpu-utils                  3.0.0-1         all          AMD GPU performance adjustment and monitoring
un  ricks-amdgpu-utils-doc              <none>          <none>       (no description available)
ii  xserver-xorg-video-amdgpu           19.1.0-1        amd64        X.Org X server -- AMDGPU display driver
ii  xserver-xorg-video-amdgpu-hwe-18.04 3:14.5          amd64        Transitional package for xserver-xorg-video-amdgpu-hwe-18.04
```
I tried to get some info on it. rocminfo apparently only finds the (Intel) CPU !?:
```
moeller@rig1c:~$ sudo /opt/rocm-4.0.0/bin/rocminfo
ROCk module is loaded
Able to open /dev/kfd read-write
=====================
HSA System Attributes
=====================
Runtime Version:         1.1
System Timestamp Freq.:  1000.000000MHz
Sig. Max Wait Duration:  18446744073709551615 (0xFFFFFFFFFFFFFFFF) (timestamp count)
Machine Model:           LARGE
System Endianness:       LITTLE

==========
HSA Agents
==========
*******
Agent 1
*******
  Name:                    Intel(R) Pentium(R) CPU G4400 @ 3.30GHz
  Uuid:                    CPU-XX
  Marketing Name:          Intel(R) Pentium(R) CPU G4400 @ 3.30GHz
  Vendor Name:             CPU
  Feature:                 None specified
  Profile:                 FULL_PROFILE
  Float Round Mode:        NEAR
  Max Queue Number:        0(0x0)
  Queue Min Size:          0(0x0)
  Queue Max Size:          0(0x0)
  Queue Type:              MULTI
  Node:                    0
  Device Type:             CPU
  Cache Info:
    L1:                      32768(0x8000) KB
  Chip ID:                 0(0x0)
  Cacheline Size:          64(0x40)
  Max Clock Freq. (MHz):   3300
  BDFID:                   0
  Internal Node ID:        0
  Compute Unit:            2
  SIMDs per CU:            0
  Shader Engines:          0
  Shader Arrs. per Eng.:   0
  WatchPts on Addr. Ranges:1
  Features:                None
  Pool Info:
    Pool 1
      Segment:                 GLOBAL; FLAGS: KERNARG, FINE GRAINED
      Size:                    32848320(0x1f539c0) KB
      Allocatable:             TRUE
      Alloc Granule:           4KB
      Alloc Alignment:         4KB
      Accessible by all:       TRUE
    Pool 2
      Segment:                 GLOBAL; FLAGS: COARSE GRAINED
      Size:                    32848320(0x1f539c0) KB
      Allocatable:             TRUE
      Alloc Granule:           4KB
      Alloc Alignment:         4KB
      Accessible by all:       TRUE
  ISA Info:
    N/A
*** Done ***
```
This is what "sudo clinfo" gives me:
```
  Platform Name                                   AMD Accelerated Parallel Processing
  Platform Vendor                                 Advanced Micro Devices, Inc.
  Platform Version                                OpenCL 2.1 AMD-APP (3004.6)
  Platform Profile                                FULL_PROFILE
  Platform Extensions                             cl_khr_icd cl_amd_event_callback cl_amd_offline_devices
  Platform Host timer resolution                  1ns
  Platform Extensions function suffix             AMD

  Platform Name                                   Clover
  Platform Vendor                                 Mesa
  Platform Version                                OpenCL 1.1 Mesa 20.0.8
  Platform Profile                                FULL_PROFILE
  Platform Extensions                             cl_khr_icd
  Platform Extensions function suffix             MESA

  Platform Name                                   AMD Accelerated Parallel Processing
Number of devices                                 2
  Device Name                                     Ellesmere
  Device Vendor                                   Advanced Micro Devices, Inc.
  Device Vendor ID                                0x1002
  Device Version                                  OpenCL 1.2 AMD-APP (3004.6)
  Driver Version                                  3004.6
  Device OpenCL C Version                         OpenCL C 1.2
  Device Type                                     GPU
  Device Board Name (AMD)                         Radeon RX 580 Series
  Device Topology (AMD)                           PCI-E, 03:00.0
  Device Profile                                  FULL_PROFILE
 ```
Hm.

When I have a bit more time at my hands again I want to help those folks who are currenly packaging ROCm for Debian which synchronizes at https://salsa.debian.org/rocm-team. Maybe this is a good place for you to continue with your quest.

**Comment by Wedge009** [comment]
Thanks for the update. Some context: I have trouble with ROCr-based OpenCL with amdgpu-pro. amdgpu-pro uses PAL or, more recently, ROCr-based OpenCL for Vega GPUs and newer. Your Polaris-based RX 580s would run on 'legacy' OpenCL (at least under amdgpu-pro) and thus has no current issues running under BOINC. I suspect my previous attempts to run ROCm as an alternative to amdgpu-pro might be related. I wonder if there is a similar 'legacy' OpenCL implementation within ROCm that allows you to run successfully, or if you are doing so only under amdgpu-pro - it looks like you have them running side-by-side?

To date, I haven't found anyone else running BOINC GPU processing with current amdgpu-pro on Vega or newer GPUs. Anyway, good on you for working on the packaging and thanks for your advice.

**Comment by ROCmSupport** [comment]
Thanks @smoe for reaching out.
From the above thread, I can always suggest that both amdgpu-pro driver and rocm driver can not exists in parallel and so request you to install one at a time.

And also, please let me know if you are still facing any issues, so that I can quickly check for you.
Request you to to share the exact steps to reproduce the problem along with the details of hardware, os etc.,.
Thank you.

**Comment by ROCmSupport** [comment]
Issue is fixed and no more observed with clean rocm 4.1 and request you try with the same.
Thank you.
